manta-accounting/src/transfer/mod.rs

```diff
@@ -1723,6 +1723,28 @@ where
         Self::new_unchecked(asset_id, sources, senders, receivers, sinks)
     }
 
+    /// Generates the public input for the [`Transfer`] validation proof.
+    #[inline]
+    pub fn generate_proof_input(self) -> ProofInput<C> {
```

Comment:
This should be the same API as `TransferPost` which borrows its input, not moves it.

Reply:
`into_post(self)` takes `self` as argument. We need to move `self` into `generate_proof_input`. Otherwise, we may need the `Clone` trait for `sender` and `receiver`.

Reply:
The `into_post` method is not the one to use. Just borrow the data that `into_post` would move.

Reply:
Remember in our conversation that you needed to add `extend_input` methods to `Sender` and `Receiver`.